### PR TITLE
Decouple SG from unmanaged driver

### PR DESCRIPTION
**Pull request by mpath**
Moves the appropriate logic to the sg driver.

JIRA:NCP-1471

### CODE REVIEW

File: quark/drivers/security_groups.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+LOG = logging.getLogger(__name__)
+
+
+class SecurityGroupDriver(object):
+
+    def __init__(self):
+        self.is_capable = self.is_security_group_capable()
+
+    def update_port(self, **kwargs):
+        if not self.is_capable:
```
Comment (asadoughi) [inline quark/drivers/security_groups.py]: ha

File: quark/drivers/security_groups.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        client = sg_client.SecurityGroupsClient(use_master=True)
+        try:
+            client.delete_vif(kwargs["device_id"],
+                              kwargs["mac_address"])
+        except Exception:
+            LOG.exception("Failed to reach the security groups backend")
+
+    @env.has_capability(env.Capabilities.SECURITY_GROUPS)
+    def is_security_group_capable(self):
+        return True
```
Comment (asadoughi) [inline quark/drivers/security_groups.py]: Memes aside, what's going on here?

Comment (mpath) [inline quark/drivers/security_groups.py]: The decorator is being undecorated. Calling the function that looks for a config value every time we call port update and in every function in the sg driver... instead I set a member var to contain the return value on __init__ and use that instead. Same effect. Should save a few cycles over the life of the product.

Comment (mpath) [inline quark/drivers/security_groups.py]: Also bike shedding/pet ducking.

Comment (asadoughi) [inline quark/drivers/security_groups.py]: I like your intent, but this is not a valuable software engineering practice. See http://c2.com/cgi/wiki?PrematureOptimization for more. Please re-factor with the decorator in place. If the decorator is shown later to be a critical performance issue, we can optimize it then.

Comment (mpath) [inline quark/drivers/security_groups.py]: It worked!

**Pull request merged**